Created simple widget

francoiscampbell opened the pull request:
Shows launcher icon, balance and last date
TODO: manual refresh button, direct settings button?

Commit:
Added simple widget
TODO: manual refresh button

davidbb commented:
thanks for this pull request. we'll take a look and merge it in. i would like it if we could refresh the balance every n minutes, or on demand by hitting the right spot on the widget. possibly the icon overlayed with a semi-transparent "refresh" icon?

francoiscampbell commented:
I've started working on the refreshing already. It works and the balance is correctly reflected both ways, I just need to make sure it performs well battery-wise. Right now I've got it setup to refresh when tapping the balance number (I know that's not very intuitive, im just trying to keep the widget 2x1, but I can make a bigger one if necessary)

I'll submit my new changes sometime tonight

Commit:
Added widget refresh

Tap balance text to refresh
TODO: Widget auto-update and redesign to make it more intuitive

davidbb commented:
I like the idea of tapping the balance number to refresh. we also probably don't need anything larger than 2x1.

davidbb commented:
hi francois. a couple of comments:

i tested the update by building your branch and installing a build on top of the play store version. and i'm getting a null pointer exception when updating the balance from the main activity. don't have time to debug, but the null pointer doesn't happen when installing your build from scratch. user should be able to upgrade cleanly, so we need to figure out what broke. however this is hard, since you also seem to have changed brackets and spacing (the diff looks like you rewrote all lines in a few files), so I can't easily tell what you changed inside the already existing files. 

in terms of the widget, it's showing up weird on my phone (see attached). it's like the font is too big. what phone/resolution are you testing on? 
![ravenwidget](https://f.cloud.github.com/assets/202203/272921/fc423780-9012-11e2-831b-0dc14c52f811.png)

i think we should have a raven balance update service that is called by both the main activity and the widget, instead of doing it 2 different ways. @binaryparadox might be able to suggest a clean way of doing it. let's merge that in first, then the widget, then we'll do the tap-the-widget-to-refresh feature.

francoiscampbell commented:
Sorry about the indent, it's Eclipse auto-format that did that. Ill turn it off when I work on this app. The widget layout is weird, it looks fine on my GS3 running CyanogenMod 10 with Action Launcher, but I went and tested it on my TF201 (also CM10, but with Nova Launcher) and two emulators: 

http://imgur.com/a/NAzHu

As for the Exception, I'm not sure, as it worked on mine wen I went from Play to Dev version. Ill roll back everything I did and I'll start by implementing the balance fetching as an IntentService instead of having a Service calling an AsyncTask. I'll close this pull request and open a new one with more "one-purpose" commits

Once this works again, how do you feel about making this app widget-only? Once it's on the homescreen, I don't see why a user would open it again, and we can have a button to launch settings, either as an Activity or even as a Dialog.

francoiscampbell committed:
Revert "Added simple widget"

This reverts commit 263fbee5c1b6bc747be66fb7eecccde876a4c871.

francoiscampbell committed:
Revert "Added widget refresh"

This reverts commit e31871da2ec5067ec553624b6ebb670ffbd540bc.

francoiscampbell committed:
Re-wrote fetcher as IntentService

davidbb commented:
thanks @francoiscampbell. i've merged your changes into a development branch. we'll use master for the play store release version. your build works fine in the emulator. me and @binaryparadox will test for a couple days then build a new play store binary. 

thanks!
